ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -245,17 +292,29 @@ private void addPutEntries(List<Long> offsets, long lastEntrySize, IndexSegment
    * @param numItems the expected numner of items the {@code indexSegment}
    * @param isMapped the expected mapped state of the {@code indexSegment}
    * @param endOffset the expected end offset of the {@code indexSegment}
+   * @param lastModifiedTimeInSecs the last modified time in secs
+   * @param resetKey the reset key for the index segment
    */
   private void verifyIndexSegmentDetails(IndexSegment indexSegment, Offset startOffset, int numItems, boolean isMapped,
-      long endOffset) {
+      long endOffset, long lastModifiedTimeInSecs, Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey) {
     String logSegmentName = startOffset.getName();
-    long indexEntrySize = KEY_SIZE + VALUE_SIZE;
+    long indexEntrySize =
+        KEY_SIZE + (version == 0 ? IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V0 : IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V1);
     assertEquals("LogSegment name not as expected", logSegmentName, indexSegment.getLogSegmentName());
     assertEquals("Start offset not as expected", startOffset, indexSegment.getStartOffset());
     assertEquals("End offset not as expected", new Offset(logSegmentName, endOffset), indexSegment.getEndOffset());
     assertEquals("Mapped state is incorrect", isMapped, indexSegment.isMapped());
     assertEquals("Key size is incorrect", KEY_SIZE, indexSegment.getKeySize());
-    assertEquals("Value size is incorrect", VALUE_SIZE, indexSegment.getValueSize());
+    assertEquals("Value size is incorrect",
+        (version == 0 ? IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V0 : IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V1),
```

Comment:
looks like you do this in two places in this function. Store as variable in the beginning?